### PR TITLE
DEM-1957 Buttons redesign

### PR DESCRIPTION
**Pull request by maria-clipp**
**What is this about?**
Redesigning the states of the buttons 

DEM-1957

**Checklist**
- [x] Upgrade storybook addons 
- [x] Move buttons related style variables in button folder
- [x] Migrate old button stories to mdx
- [x] Adding start and end icon props 
- [x] DIsabled style for clippings and kazam 
- [x] Request 1st reviewer early
- [x] Request 2nd reviewer

NOTES:
- Updating the add ons was necessary because the mdx stories were not working correctly.
- Out of scope, I have added `startIcon` and `endIcon` props to the buttons that accept components if you feel like it is unnecessary I will remove it.

### CODE REVIEW

File: src/stories/Accordion/Accordion.stories.tsx [hunk 1]
```diff
@@ -1,40 +0,0 @@
-import React from 'react';
```
Comment (MartinAndreev) [inline src/stories/Accordion/Accordion.stories.tsx:1]: Is there a reason this is removed?

Comment (maria-clipp) [reply src/stories/Accordion/Accordion.stories.tsx:1]: No this is a mistake. Thank you for catching it.

**Comment by MartinAndreev** [comment]
As we talked in Slack, i am posting everything here so it is visible:

If we remove the notes addon it should be removed from the addons as well.
Also if we migrate to newer plugins, we can fix the config a little, i was workin on that in my free time to update all the docs, but as you did part of it i wll just copy the config here.


```javascript
// main.js
addons: ['@storybook/addon-essentials', '@storybook/addon-actions', '@storybook/addon-links', '@storybook/addon-viewport', '@storybook/addon-knobs', 'storybook-addon-jsx'],
```

```javascript
// preview.js
import { addDecorator, addParameters } from '@storybook/react';
import { withConsole } from '@storybook/addon-console';
import '@storybook/addon-console';

addDecorator((storyFn, context) => withConsole()(storyFn)(context));
```

and you can delete the addons.js as it will be removed later

**Comment by maria-clipp** [comment]
@AngelGenchev95 @MartinAndreev @EZarkov @iliyan-h  please when you have time take a look at the pr.

**Review by EZarkov — APPROVED** [review]

**Review by AngelGenchev95 — APPROVED** [review]

**Pull request merged**